src/Sylius/Component/Core/StateResolver/OrderStateResolver.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\StateResolver;
+
+use SM\Factory\FactoryInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Sylius\Component\Core\OrderShippingStates;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Order\OrderTransitions;
+use Sylius\Component\Order\StateResolver\StateResolverInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class OrderStateResolver implements StateResolverInterface
```

Comment:
Is it a good name? This service does not resolve order state.. it just fulfills it if its possible :|